### DISCUSSION TITLE
Insert list of Paire<Number,Number>

### DISCUSSION THREAD

**Discussion by ThomasLefebvre** [patrykandpatrick/vico#412] [Ideas]
hi Patryk and Patrick, I propose to create a function that allows to insert directly a list of Pair<Number,Number>, it has been very used in my case.

Thanks in advance,
```
fun entryModelOfList(vararg entries: List<Pair<Number, Number>>): ChartEntryModel =
    entries
        .map {
            it.map { (x, y) ->
                entryOf(x.toFloat(), y.toFloat())
            }
        }
        .let { entryList -> ChartEntryModelProducer((entryList)) }
        .getModel()
```

**Comment by patrickmichalik** [comment]
Hello! Thanks for the suggestion. The currently available utility functions enable you to achieve the same result in more or less the same number of characters. Please compare the following. The first snippet uses your function, and the second one uses two existing Vico functions.

```kt
entryModelOfList(listOf(0f to 2f, 2f to 4f), listOf(0f to 4f, 2f to 4f))
```

```kt
entryModelOf(entriesOf(0f to 2f, 2f to 4f), entriesOf(0f to 4f, 2f to 4f))
```

Please let me know if I’m missing something, but this leads me to believe that `entryModelOfList` may not be necessary.

**Reply by ThomasLefebvre** [reply]
Hello, thank you for your answers,

Unfortunately with your solution I couldn't find a way to insert directly a list received from the api.

For example : 

`val modelData = entryModelOfList(listDataElevation, listDataCadence)`

I therefore leave the function local for my needs.

Thx Thomas

**Reply by patrickmichalik** [reply]
Ah, got it. If you were to use arrays instead of lists, you could utilize the spread operator:

```kt
entryModelOf(entriesOf(*series1), entriesOf(*series2))
```

Because Kotlin doesn’t have a spread operator for lists, your `entryModelOfList` function is the best bet otherwise.

The current selection of utility functions covers most simple use cases, which was the aim for these functions. It could be argued that some of them are already on the specialized side. Because there are dozens of possible use cases, we’d like to keep the API relatively light, and creating such functions is relatively simple, we currently don’t plan on expanding this list of utility functions. We suggest that you keep `entryModelOfList` in your project—it’s perfect for your use case. Cheers!
